libs/utils/src/simple_rcu.rs

```diff
@@ -9,6 +9,36 @@
 //! This implementation isn't wait-free; it uses an RwLock that is held for a
 //! short duration when the value is read or updated.
 //!
+//! # Examples
+//!
+//! Read a value and do things with it while holding the guard:
+//!
+//! ```
+//! # let rcu = utils::simple_rcu::Rcu::new(1);
+//! {
+//!     let read = rcu.read();
+//!     println!("the current value is {}", *read);
+//!     // exiting the scope drops the read-guard, and allows concurrent writers
+//!     // to finish.
+//! }
+//! ```
+//!
+//! Increment the value by one, and wait for old readers to finish:
+//!
+//! ```
+//! # let rcu = utils::simple_rcu::Rcu::new(1);
+//! let write_guard = rcu.write();
+//!
+//! // NB: holding `write_guard` blocks new readers and writers. Keep this section short!
+//! let new_value = *write_guard + 1;
+//!
+//! let waitlist = write_guard.store(new_value); // consumes `write_guard`
```

Comment:
IMO this comment "consumes `write_guard`" is doing the most heavy lifting and will probably be copy-pasted in all uses of `store`. It's obvious when you look at the type signature that store eats the write_guard, but not obvious if you're just reading the code.

Maybe renaming `store` to `unlock_and_store` would improve things? I don't have a better suggestion, feel free not to take this one

Reply:
> IMO this comment "consumes `write_guard`" is doing the most heavy lifting and will probably be copy-pasted in all uses of `store`. It's obvious when you look at the type signature that store eats the write_guard, but not obvious if you're just reading the code.

Hmm, agreed.

> Maybe renaming `store` to `unlock_and_store` would improve things? I don't have a better suggestion, feel free not to take this one

That alone would be asymmetric; if there's "unlock" there should also be "lock". Perhaps `lock_for_write()` and `store_and_unlock()`.

Another approach would be to pass a closure to the function that gets executed while holding the lock:
```
let waitlist = rcu.write(|old_value| old_value + 1);
waitlist.wait();
```
One nice thing about a closure is that it encourages to keep the locked section short. It's easier to write code that passes around a guard-object and holds onto it for a long time, than code with a large closure.

A downside is that it's more cumbersome to return the old or the new value to the calling function, if you need it. It would need to be something like this:
```
let mut old_val = None;
let waitlist = rcu.write(|old_value| { old_val = Some(old_value); old_value + 1});
waitlist.wait();
```
The current only caller of this has a similar problem with returning an error. That's cumbersome with a closure.

So, I renamed it to `lock_for_write` and `store_and_unlock`.